src/sentry/db/models/manager.py

```diff
@@ -317,6 +317,32 @@ def get_queryset(self):
         return self._queryset_class(self.model, using=self._db)
 
 
+class SnubaEventManager(BaseManager):
+    def from_event_id(self, id_or_event_id, project_id):
+        """
+        Get a SnubaEvent by either its id primary key or its hex event_id.
+
+        Returns None if the event cannot be found under either scheme.
+
+        Log any attempt to fetch a SnubaEvent by primary key and eventually remove.
+        """
+        from sentry.models import SnubaEvent, Event
+
+        if not is_event_id(id_or_event_id):
+            logger.warning('Attempt to fetch SnubaEvent by primary key', exc_info=True, extra={
+                'stack': True
+            })
+
+            event = Event.objects.from_event_id(id_or_event_id, project_id)
+
+            if not event:
+                return None
+
+            id_or_event_id = event.event_id
+
+        return SnubaEvent.get_event(project_id, id_or_event_id)
```

Comment:
We could in theory move the code for `get_event` into this method as well I guess, but it's probably fine as it is for now.